src/neo/SmartContract/ApplicationEngine.OpCodePrices.cs

```diff
@@ -7,190 +7,190 @@ partial class ApplicationEngine
-            [OpCode.PUSH4] = 30,
-            [OpCode.PUSH5] = 30,
-            [OpCode.PUSH6] = 30,
-            [OpCode.PUSH7] = 30,
-            [OpCode.PUSH8] = 30,
-            [OpCode.PUSH9] = 30,
-            [OpCode.PUSH10] = 30,
-            [OpCode.PUSH11] = 30,
-            [OpCode.PUSH12] = 30,
-            [OpCode.PUSH13] = 30,
-            [OpCode.PUSH14] = 30,
-            [OpCode.PUSH15] = 30,
-            [OpCode.PUSH16] = 30,
-            [OpCode.NOP] = 30,
-            [OpCode.JMP] = 70,
-            [OpCode.JMP_L] = 70,
-            [OpCode.JMPIF] = 70,
-            [OpCode.JMPIF_L] = 70,
-            [OpCode.JMPIFNOT] = 70,
-            [OpCode.JMPIFNOT_L] = 70,
-            [OpCode.JMPEQ] = 70,
-            [OpCode.JMPEQ_L] = 70,
-            [OpCode.JMPNE] = 70,
-            [OpCode.JMPNE_L] = 70,
-            [OpCode.JMPGT] = 70,
-            [OpCode.JMPGT_L] = 70,
-            [OpCode.JMPGE] = 70,
-            [OpCode.JMPGE_L] = 70,
-            [OpCode.JMPLT] = 70,
-            [OpCode.JMPLT_L] = 70,
-            [OpCode.JMPLE] = 70,
-            [OpCode.JMPLE_L] = 70,
-            [OpCode.CALL] = 22000,
-            [OpCode.CALL_L] = 22000,
-            [OpCode.CALLA] = 22000,
-            [OpCode.ABORT] = 30,
-            [OpCode.ASSERT] = 30,
-            [OpCode.THROW] = 22000,
-            [OpCode.TRY] = 100,
-            [OpCode.TRY_L] = 100,
-            [OpCode.ENDTRY] = 100,
-            [OpCode.ENDTRY_L] = 100,
-            [OpCode.ENDFINALLY] = 100,
+            [OpCode.PUSHINT8] = 1 << 0,
+            [OpCode.PUSHINT16] = 1 << 0,
+            [OpCode.PUSHINT32] = 1 << 0,
+            [OpCode.PUSHINT64] = 1 << 0,
+            [OpCode.PUSHINT128] = 1 << 2,
+            [OpCode.PUSHINT256] = 1 << 2,
+            [OpCode.PUSHA] = 1 << 2,
+            [OpCode.PUSHNULL] = 1 << 0,
+            [OpCode.PUSHDATA1] = 1 << 3,
+            [OpCode.PUSHDATA2] = 1 << 9,
+            [OpCode.PUSHDATA4] = 1 << 12,
+            [OpCode.PUSHM1] = 1 << 0,
+            [OpCode.PUSH0] = 1 << 0,
+            [OpCode.PUSH1] = 1 << 0,
+            [OpCode.PUSH2] = 1 << 0,
+            [OpCode.PUSH3] = 1 << 0,
+            [OpCode.PUSH4] = 1 << 0,
+            [OpCode.PUSH5] = 1 << 0,
+            [OpCode.PUSH6] = 1 << 0,
+            [OpCode.PUSH7] = 1 << 0,
+            [OpCode.PUSH8] = 1 << 0,
+            [OpCode.PUSH9] = 1 << 0,
+            [OpCode.PUSH10] = 1 << 0,
+            [OpCode.PUSH11] = 1 << 0,
+            [OpCode.PUSH12] = 1 << 0,
+            [OpCode.PUSH13] = 1 << 0,
+            [OpCode.PUSH14] = 1 << 0,
+            [OpCode.PUSH15] = 1 << 0,
+            [OpCode.PUSH16] = 1 << 0,
+            [OpCode.NOP] = 1 << 0,
+            [OpCode.JMP] = 1 << 1,
+            [OpCode.JMP_L] = 1 << 1,
+            [OpCode.JMPIF] = 1 << 1,
+            [OpCode.JMPIF_L] = 1 << 1,
+            [OpCode.JMPIFNOT] = 1 << 1,
+            [OpCode.JMPIFNOT_L] = 1 << 1,
+            [OpCode.JMPEQ] = 1 << 1,
```

Comment:
JMPs with embedded comparisons also reliably use more execution time than simple JMPs, not a lot, but enough to justify `1<<2` vs `1<<1`.

Reply:
Should be good for JMPEQ, JMPEQ_L, JMPNE and JMPNE_L